src/runtime/long.cpp

```diff
@@ -397,7 +396,6 @@ Box* longStr(BoxedLong* v) {
 
     char* buf = mpz_get_str(NULL, 10, v->n);
     auto rtn = new BoxedString(buf);
-    free(buf);
```

Comment:
Hmm, could we have longStr call malloc like longRepr?  It would make this behavior not dependent on whether or not we're using the custom allocator.